File: bank-extract-api/src/main/java/br/com/bank/bank/bankextractapi/adapter/in/web/mapper/ActivityWebMapper.java
package br.com.bank.bank.bankextractapi.adapter.in.web.mapper;

import br.com.bank.bank.bankextractapi.adapter.in.web.dto.ActivityDto;
import br.com.bank.bank.bankextractapi.domain.Activity;
import org.springframework.stereotype.Component;

@Component
public class ActivityWebMapper {

    public Activity toDomain(ActivityDto activityDto) {
        return new Activity(activityDto.getId(), activityDto.getSourceAccount(), activityDto.getTargetAccount(), Activity.ActivityType.valueOf(activityDto.getType()), activityDto.getValue(), activityDto.getBalance(), activityDto.getDate());
    }
}

File: bank-api/src/main/kotlin/br/com/bank/adapter/in/web/mapper/ActivityWebMapper.kt
package br.com.bank.adapter.`in`.web.mapper

import br.com.bank.adapter.`in`.web.dto.ActivityDto
import br.com.bank.domain.Activity
import org.springframework.stereotype.Component

@Component
class ActivityWebMapper {

    fun toDto(activity: Activity): ActivityDto = ActivityDto(activity.id!!, activity.sourceAccount?.name, activity.targetAccount?.name, activity.type.name, activity.value, activity.balance, activity.date)
}